src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponentMastery.cs

```diff
@@ -155,6 +196,21 @@ protected override void Apply(DomainEvent @event)
             When((dynamic)@event);
         }
 
+        private void When(SessionLaunched @event)
+        {
+            HasActiveSession = true;
+        }
+
+        private void When(SessionTerminated @event)
+        {
+            HasActiveSession = false;
+        }
+
+        private void When(SessionAbandoned @event)
+        {
+            HasActiveSession = false;
+        }
+
         private void When(AssessmentEventAnswered @event)
         {
             /* TODO: refactor to move the GetMaximumSubmissionCorrectness from AE to KCMastery, or a 
```

Comment:
Can you remove these TODOs from the code? You can leave the comments as is and I'll create a task for enhancing the KCM aggregate after the empirical evaluation, once we analyze the data and understand in which direction the KCM should be moving in.